internals/overlord/cmdstate/request.go

```diff
@@ -117,12 +117,15 @@ func Exec(st *state.State, args *ExecArgs) (*state.Task, ExecMetadata, error) {
 		environment["LANG"] = "C.UTF-8"
 	}
 
-	// Set default working directory to $HOME, or / if $HOME not set.
+	// Set default working directory to $HOME, or / if $HOME not set or doesn't exist.
 	workingDir := args.WorkingDir
 	if workingDir == "" {
 		workingDir = environment["HOME"]
 		if workingDir == "" {
 			workingDir = "/"
+		} else if !osutil.IsDir(workingDir) {
+			logger.Debugf("HOME dir %q does not exist, using /", workingDir)
+			workingDir = "/"
```

Comment:
Actually, I think this is very dangerous. We should warn, or fail, but never change the working directory.

If WorkingDir is set we must use that, if HOME is set we must use that if neither are set we must use `/`.
If I run a command `rm -rf bin` with HOME=/home/ubuntu, semantically I want to remove the bin directory in /home/ubuntu. We should never default that to `/`. The user was explicit.

Reply:
Per discussion, I've updated the PR to continue to fail in this case, but return a much better error message.